src/m365/aad/commands/user/user-license-list.ts

```diff
@@ -0,0 +1,106 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { odata } from '../../../../utils/odata';
+import { validation } from '../../../../utils/validation';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId?: string;
+  userName?: string;
+}
+
+class AadUserLicenseListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.USER_LICENSE_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists the license details for a given user';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'skuId', 'skuPartNumber'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId as string)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({
+      options: ['userId', 'userName'],
+      runsWhen: (args) => {
+        const options = [args.options.userId, args.options.userName];
+        return options.some(item => item !== undefined);
+      }
+    });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let requestUrl: string = `${this.resource}/v1.0/`;
```

Comment:
Let's also add a validation that people should specify any of the options when they are using application permissions.